sqlite-android/src/main/jni/sqlite/android_database_SQLiteConnection.cpp

```diff
@@ -827,14 +829,18 @@ static void nativeLoadExtension(JNIEnv* env, jobject clazz,
         if (proc) {
             procChars = env->GetStringUTFChars(proc, NULL);
         }
-        result = sqlite3_load_extension(connection->db, fileChars, procChars, 0);
+        result = sqlite3_load_extension(connection->db, fileChars, procChars, &errorMessage);
         env->ReleaseStringUTFChars(file, fileChars);
         if (proc) {
             env->ReleaseStringUTFChars(proc, procChars);
         }
     }
     if (result != SQLITE_OK) {
+        char* formattedError = sqlite3_mprintf("Could not register extension: %s", errorMessage);
+        sqlite3_free(errorMessage);
+
         throw_sqlite3_exception_errcode(env, result, "Could not register extension");
```

Comment:
Looks like you want to pass `formattedError` into `throw_sqlite3_exception_errcode` so the exception has the message in it?

Reply:
Exactly - I found it necessary to provide more verbose error messages to figure out what was wrong with a module I was trying to load. It was helpful to me, so I figured it could be helpful to push upstream.

Reply:
It looks like I may have forgotten to commit the bit that actually *uses* the formatted string.